cmake/modules/dts.cmake

```
@@ -125,17 +125,28 @@ set(VENDOR_PREFIXES             dts/bindings/vendor-prefixes.txt)
 
 # TODO: What to do about non-posix platforms where NOT CONFIG_HAS_DTS (xtensa)?
 # Drop support for NOT CONFIG_HAS_DTS perhaps?
-set_ifndef(DTS_SOURCE ${BOARD_DIR}/${BOARD}.dts)
-if(EXISTS ${DTS_SOURCE})
-  # We found a devicetree. Check for a board revision overlay.
-  if(BOARD_REVISION AND EXISTS ${BOARD_DIR}/${BOARD}_${BOARD_REVISION_STRING}.overlay)
-    list(APPEND DTS_SOURCE ${BOARD_DIR}/${BOARD}_${BOARD_REVISION_STRING}.overlay)
-  endif()
+set(BOARD_DTS ${BOARD_DIR}/${BOARD}.dts)
+set(FOUND_BOARD_DTS FALSE)
+set(HAD_DTS_SOURCE FALSE)
+if(DEFINED DTS_SOURCE)
+  message(STATUS "Using devicetree: ${DTS_SOURCE}")
+  set(HAD_DTS_SOURCE TRUE)
+  set(ORIGINAL_DTS_SOURCE "${DTS_SOURCE}")
 else()
-  # If we don't have a devicetree after all, there's not much to do.
-  set(header_template ${ZEPHYR_BASE}/misc/generated/generated_header.template)
-  zephyr_file_copy(${header_template} ${DEVICETREE_GENERATED_H} ONLY_IF_DIFFERENT)
-  return()
+  if(EXISTS ${BOARD_DTS})
+    message(STATUS "Found BOARD.dts: ${BOARD_DTS}")
+    set(FOUND_BOARD_DTS TRUE)
+    set(DTS_SOURCE ${BOARD_DTS})
+    # We found a devicetree. Check for a board revision overlay.
+    if(BOARD_REVISION AND EXISTS ${BOARD_DIR}/${BOARD}_${BOARD_REVISION_STRING}.overlay)
+      list(APPEND DTS_SOURCE ${BOARD_DIR}/${BOARD}_${BOARD_REVISION_STRING}.overlay)
```

If adding a message print here, then we can remove all the extra book keeping as any file present in `DTS_SOURCE` has been print when reaching code below:
```suggestion
    set(board_rev_overlay ${BOARD_DIR}/${BOARD}_${BOARD_REVISION_STRING}.overlay)
    if(BOARD_REVISION AND EXISTS ${board_rev_overlay})
      message(STATUS "Found devicetree overlay: ${board_rev_overlay}")
      list(APPEND DTS_SOURCE ${board_rev_overlay})
```